app/client/components/consent/PrivacySettings.tsx

```diff
@@ -232,25 +314,55 @@ export class PrivacySettings extends Component<{}, State> {
     }
 
     return new Promise(resolve =>
-      this.setState({ iabPurposes }, () => resolve())
+      this.setState({ guPurposes, iabPurposes }, () => resolve())
+    );
+  }
+
+  private renderGuPurposeItems(): React.ReactNode {
+    if (!this.guPurposeList || !this.guPurposeList.purposes) {
+      return "";
+    }
+
+    return this.guPurposeList.purposes.map(
+      (purpose: ParsedGuPurpose): React.ReactNode => {
+        const { id, name, description, integDescription } = purpose;
+
+        // Remove radio button for Essential
```

Comment:
I like your use of `optProps` but I'm not totally sure about the `id === 0` check, I know the comment above explains what's going on but comments are easy to remove. I'd  much rather a more descriptive variable name was used in place of  `id === 0` to  demonstrate what's going on here. Either `const  hideRadio = id === 0` or  stick a boolean `hideRadio` on the `purpose`.

Reply:
👍